nemo/collections/asr/models/hybrid_rnnt_ctc_bpe_models.py

```diff
@@ -0,0 +1,185 @@
+            if self.tokenizer_type == "agg":
+                new_decoder_config.vocabulary = ListConfig(vocabulary)
+            else:
+                new_decoder_config.vocabulary = ListConfig(list(vocabulary.keys()))
+            new_decoder_config['num_classes'] = len(vocabulary)
+
+            self.ctc_decoder = EncDecCTCModelBPE.from_config_dict(new_decoder_config)
+            self.ctc_loss = CTCLoss(
+                num_classes=self.ctc_decoding.num_classes_with_blank - 1,
+                zero_infinity=True,
+                reduction=self.cfg.ctc_decoding.get("ctc_reduction", "mean_batch"),
+            )
+
+            if ctc_decoding_cfg is None:
+                ctc_decoding_cfg = self.cfg.ctc_decoding.get('decoding', None)
+            if ctc_decoding_cfg is None:
+                ctc_decoding_cfg = OmegaConf.structured(CTCBPEDecodingConfig)
+                with open_dict(self.cfg.ctc_decoding):
+                    self.cfg.ctc_decoding.decoder = ctc_decoding_cfg
+
+            # Assert the decoding config with all hyper parameters
+            ctc_decoding_cls = OmegaConf.structured(CTCBPEDecodingConfig)
+            ctc_decoding_cls = OmegaConf.create(OmegaConf.to_container(ctc_decoding_cls))
+            ctc_decoding_cfg = OmegaConf.merge(ctc_decoding_cls, ctc_decoding_cfg)
+
+            self.ctc_decoding = CTCBPEDecoding(decoding_cfg=ctc_decoding_cfg, tokenizer=self.tokenizer)
+
+            self.ctc_wer = WERBPE(
+                decoding=self.ctc_decoding,
+                use_cer=self.cfg.ctc_decoding.get('use_cer', False),
+                log_prediction=self.cfg.get("log_prediction", False),
+                dist_sync_on_step=True,
+            )
+
+            # Update config
+            with open_dict(self.cfg.ctc_decoding.decoder):
+                self.cfg.ctc_decoding.decoder = new_decoder_config
+
+            with open_dict(self.cfg.ctc_decoding.decoding):
+                self.cfg.ctc_decoding.decoding = ctc_decoding_cfg
+
+    def change_decoding_strategy(self, decoding_cfg: DictConfig, decoder_type: str = None):
+        """
+        Changes decoding strategy used during RNNT decoding process.
+        Args:
+            decoding_cfg: A config for the decoder, which is optional. If the decoding type
+                needs to be changed (from say Greedy to Beam decoding etc), the config can be passed here.
+            decoder_type: (str) Can be set to 'rnnt' or 'ctc' to switch between appropriate decoder in a
+                model having both RNN-T and CTC decoders. Defaults to None, in which case RNN-T decoder is
+                used. If set to 'ctc', it raises error if 'ctc_decoder' is not an attribute of the model.
+        """
+        if decoder_type is None or decoder_type == 'rnnt':
+            super().__init__(decoding_cfg=decoding_cfg)
+        else:
+            assert decoder_type == 'ctc' and hasattr(self, 'ctc_decoding')
```

Comment:
Use if instead of assert and raise value error explaining the ctc decoding config is not available.
Actually maybe you can just load up the default CTCBPEDecoding config and not raise an error ? Upto you

Reply:
updated.